pkg/k8s/informers.go

```diff
@@ -144,7 +148,7 @@ func WithConfigClient(configClient configClientset.Interface) ClientOption {
 			opt.FieldSelector = fields.OneTermEqualSelector(metav1.ObjectNameField, c.meshConfigName).String()
 		})
 		meshConfiginformerFactory := configInformers.NewSharedInformerFactoryWithOptions(configClient, DefaultKubeEventResyncInterval, configInformers.WithNamespace(c.osmNamespace), listOption)
-		mrcInformerFactory := configInformers.NewSharedInformerFactoryWithOptions(configClient, DefaultKubeEventResyncInterval, configInformers.WithNamespace(c.osmNamespace))
+		mrcInformerFactory := configInformers.NewSharedInformerFactoryWithOptions(configClient, MRCResyncInterval, configInformers.WithNamespace(c.osmNamespace))
```

Comment:
@keithmattix  what're your thoughts on adding this to the event handler instead of the underlying informer? That way if anything else wants to use the informer it is consistent with the user across other informers.

This would mean exposing the `AddEventHandlerWithResyncPeriod` instead of the AddEventHandler that is called in the MRCComposer's Watch method

Reply:
Hmm good question....is there any use of the MRC informer that shouldn't have this resync?

Reply:
i think consistency across these is more important, so a future feature doesn't get surprised. surprises are bad

Reply:
Not sure that constitutes a "surprise"; in my opinion, the bigger surprise would be that the behavior of a similar feature (e.g. MRC) is different than a future feature. While one should generally try to avoid surprises, I feel like "surprises are bad" overly simplifies the nuance that accompanies these kinds of conversations

Reply:
>That way if anything else wants to use the informer it is consistent with the user across other informers.

are you suggesting leaving this at zero except if we add a second handler as we are doing here?  

If we are we using our "own ticker" (not sure where this is implemented) is changing the default resync going to conflict?

Reply:
I'm suggesting to keep the informers resync at 0 to keep consistency (I would be surprised if I had used 10+ other of our informer uses, all with no resync, but then i have a use case for that one, and it has a resync).

But instead add a method that exposes the AddEventHandlerWithResyncPeriod, so the caller is aware of the resync and explicitly states it when adding the handler.

Reply:
makes sense to me

Reply:
Cool, I don't care too much one way or another.

Reply:
@steeling @shalier @keithmattix This ends up resulting in `The specified resyncPeriod 2m30s is invalid because this shared informer doesn't support resyncing`.  

I tested it out and the base informer needs to have a resync enabled (aka a value > 1s) otherwise it resync is disabled. https://github.com/kubernetes/client-go/blob/c81636cd4415f0858a99c2cc18023740df7aa15f/tools/cache/shared_informer.go#L516-L524

We've been running the tests in https://github.com/openservicemesh/osm/pull/5201 without resync so I don't know that we need it?

Reply:
yes resync was only necessary when we were doing time based changes